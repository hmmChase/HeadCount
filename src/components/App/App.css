/*APP COMPONENT RELATED CSS GOES HERE*/
div {    
  text-align: center;
  font-family: 'Crimson Text', serif;
  background-attachment: fixed;
  background-image: url('http://backgroundcheckall.com/wp-content/uploads/2017/12/white-gray-background-5.jpg')
}

#root {
  min-height: 100vh;
}

body::-webkit-scrollbar {
  width: 1em;
}

body::-webkit-scrollbar-track {
  -webkit-box-shadow: inset 0 0 15px rgba(0,0,0,0.3);
}

body::-webkit-scrollbar-thumb {
background-color: #416387;
outline: 1px solid #416387;
}