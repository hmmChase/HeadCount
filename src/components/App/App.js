import React, { Component } from 'react';
import DistrictRepository from '../../helper';
import kinderData from '../../data/kindergartners_in_full_day_program';
import Form from '../Form/Form';
import ComparisonContainer from '../ComparisonContainer/ComparisonContainer';
import DistrictContainer from '../DistrictContainer/DistrictContainer';

import './App.css';

class App extends Component {
  constructor() {
    super();

    this.state = {
      district: new DistrictRepository(kinderData),
      foundDistricts: [],
      comparedDistricts: [],
      comparisonObject: {}
    };
  }

  componentDidMount() {
    const foundDistricts = this.state.district.findAllMatches();
    this.setState({
      foundDistricts
    });
  }

  filterDistricts = searchValue => {
    const filteredDistricts = this.state.district.findAllMatches(searchValue);
    this.setState({
      foundDistricts: filteredDistricts
    });
  };

  changeSelectedDistricts = district => {
    const selectedDistrictObject = this.state.district.findByName(district);

    const duplicate = this.state.comparedDistricts.includes(
      selectedDistrictObject
    );
    
    if (duplicate) {
      const filteredDistricts = this.state.comparedDistricts.filter(
        comparedDistrict => comparedDistrict !== selectedDistrictObject
      );
      this.updateComparedDistricts(filteredDistricts);

    } else if (this.state.comparedDistricts.length === 1) {
      const addedDistrict = [
        ...this.state.comparedDistricts,
        selectedDistrictObject
      ];
      this.setState({
        comparedDistricts: addedDistrict
      }, () => this.runComparison());
      
    } else if (this.state.comparedDistricts.length === 2) {
      const replacedDistrict = [
        this.state.comparedDistricts[0],
        selectedDistrictObject
      ];
      this.setState({
        comparedDistricts: replacedDistrict
      }, () => this.runComparison());
      
    } else {
      const addedDistrict = [
        ...this.state.comparedDistricts,
        selectedDistrictObject
      ];
      this.updateComparedDistricts(addedDistrict);
    }
  };

  updateComparedDistricts = newState => {
    this.setState({
      comparedDistricts: newState
    });
  };

  runComparison = () => {
    let comparedObject = this.state.district.compareDistrictAverages(this.state.comparedDistricts[0].location, this.state.comparedDistricts[1].location);
    this.setState({
      comparisonObject: comparedObject
    });
  }

  render() {
    return (
      <div>
        <Form filterDistricts={this.filterDistricts} />
        <ComparisonContainer 
          comparedDistricts={this.state.comparedDistricts}
          changeSelectedDistricts={this.changeSelectedDistricts}
          comparisonObject={this.state.comparisonObject}
        />
        <DistrictContainer
          foundDistricts={this.state.foundDistricts}
          changeSelectedDistricts={this.changeSelectedDistricts}
        />
      </div>
    );
  }
}

export default App;
