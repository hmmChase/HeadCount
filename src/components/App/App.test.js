import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import kinderData from '../../data/kindergartners_in_full_day_program';

import {shallow} from 'enzyme'

describe('App', () =>{
  let app

  beforeEach(() => {
    app = shallow(<App />, { disableLifecycleMethods: true })
  })
  
  it('matches snapshot', () => {
    expect(app).toMatchSnapshot()
  })

  describe('componentDidMount', () => {
    it('sets app state', () => {
      app.instance().componentDidMount()

      expect(app.state().foundDistricts).toHaveLength(181)
    })
  })

  describe('filterDistricts', () => {
    it('filters districts and is not case sensitive', () => {
      app.instance().filterDistricts('colo')

      expect(app.state().foundDistricts).toHaveLength(2)
    })
  })
})
