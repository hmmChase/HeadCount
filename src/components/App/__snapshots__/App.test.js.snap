// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Welcome To Headcount 2.0
</h1>,
        <Form
          filterDistricts={[Function]}
/>,
        <DistrictContainer
          foundDistricts={Array []}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome To Headcount 2.0",
        },
        "ref": null,
        "rendered": "Welcome To Headcount 2.0",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "filterDistricts": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "foundDistricts": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Welcome To Headcount 2.0
</h1>,
          <Form
            filterDistricts={[Function]}
/>,
          <DistrictContainer
            foundDistricts={Array []}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome To Headcount 2.0",
          },
          "ref": null,
          "rendered": "Welcome To Headcount 2.0",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "filterDistricts": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "foundDistricts": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
